subprojects/testfx-core/src/test/java/org/testfx/util/DebugUtilsTest.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    @Test
+    public void showPressedKeysInErrorMessage() throws TimeoutException {
+        // given:
+        Stage stage = FxToolkit.toolkitContext().getRegisteredStage();
+        moveTo(stage)
+            .moveBy(10, 10)
+            .press(KEY);
+
+        // when:
+        AssertionError error = getThrownErrorPostMapper(showKeysPressedAtTestFailure(this, TAB));
+
+        // then:
+        assertThat(error.getMessage(), hasAddedTextThatContains(KEY.name()));
+    }
+
+    @Test
+    public void showFiredEventsInErrorMessage() throws TimeoutException {
+        // given:
+        Stage stage = FxToolkit.setupStage(stage0 -> stage0.setScene(new Scene(new Region(), 300, 300)));
+        interact(stage::show);
+        moveTo(stage).moveBy(20, 20);
+
+        // and:
+        FiredEvents events = FiredEvents.beginStoringFiredEventsOf(stage);
+        moveBy(1, 0);
+        events.stopStoringFiredEvents();
+
+        // when:
+        AssertionError error = getThrownErrorPostMapper(showFiredEvents(events, TAB));
+
+        // then:
+        assertThat(error.getMessage(), hasAddedTextThatContains("MouseEvent"));
+        assertThat(error.getMessage(), hasAddedTextNotContaining("ScrollEvent"));
+    }
+
+    private AssertionError getThrownErrorPostMapper(Function<StringBuilder, StringBuilder> errorMessageMapper) {
+        try {
+            FxAssert.verifyThat(false, alwaysFail(), errorMessageMapper);
+            throw new AssertionError("Error: verifyThat did not throw an error when it should have.");
+        }
+        catch (AssertionError error) {
+            return error;
+        }
+    }
+
+    private Matcher<Object> alwaysFail() {
+        return new AlwaysFailsMatcher();
+    }
+
+    private Matcher<String> hasAddedTextEqualTo(String addedText) {
+        return new RemoveBlankMessageMatcher(addedText) {
+            @Override
+            public boolean matches(Object object) {
+                return addedText.equals(replaceBlankPart(object));
+            }
+        };
+    }
+
+    private Matcher<String> hasAddedTextThatContains(String text) {
+        return new RemoveBlankMessageMatcher(
+            text,
+            addedText -> "Added text: '" + addedText + "' does not contain '" + text + "'"
+        ) {
+
+            @Override
+            public boolean matches(Object item) {
+                return replaceBlankPart(item).contains(text);
+            }
+        };
+    }
+
+    private Matcher<String> hasAddedTextNotContaining(String text) {
+        return new RemoveBlankMessageMatcher(
+            text,
+            addedText -> "Added text: '" + addedText + "' contains '" + text + "'"
+        ) {
+
+            @Override
+            public boolean matches(Object item) {
+                return !replaceBlankPart(item).contains(text);
+            }
+        };
+    }
+
+    private abstract class RemoveBlankMessageMatcher extends BaseMatcher<String> {
```

Comment:
I just realized that this isn't actually needed. Why? Because there's a much easier way to remove the blank part by taking any function `f` and calling `f.compose(sb -> new StringBuilder())`